hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java

```diff
@@ -282,11 +282,7 @@ private Nft buildNewlyMintedNft(
     @Override
     public Fees calculateFees(@NonNull final FeeContext feeContext) {
         final var op = feeContext.body().tokenMintOrThrow();
-        final var readableTokenStore = feeContext.readableStore(ReadableTokenStore.class);
-        final var tokenType = readableTokenStore.get(op.tokenOrThrow()).tokenType();
-        final var subType = tokenType == TokenType.FUNGIBLE_COMMON
-                ? SubType.TOKEN_FUNGIBLE_COMMON
-                : SubType.TOKEN_NON_FUNGIBLE_UNIQUE;
+        final var subType = op.amount() > 0 ? SubType.TOKEN_FUNGIBLE_COMMON : SubType.TOKEN_NON_FUNGIBLE_UNIQUE;
```

Comment:
I don't get it.  Old code was a good way to determine fungible/non-fungible: just look directly at the token type, it'll tell you.  This code goes back to the way we see all over the mono service - figuring it out indirectly from other cues, like amount > 0.  Which in the past has led to problems, which made it to production.  

Why make this change?

Reply:
Because when we pass [INVALID_TOKEN_ADDRESS](https://github.com/hashgraph/hedera-services/blob/119c1cc180ab7e1ec1a0143c68dd1d49275f42b5/hedera-node/hedera-app/src/xtest/java/contract/MintsXTest.java#L128) in the X test the old code throws null pointer exception and can not get [tokenType()](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java#L288)

Reply:
Then maybe that case should be explicitly tested for and handled?  So it is _explicit_ in the code what is happening, that there's a necessary check for an invalid token address?  Rather than just papering it over so it's hidden?